htsget.md

```diff
@@ -41,9 +41,13 @@ JSON responses SHOULD include a `Content-Type` header describing the htsget prot
 
     Content-Type: application/vnd.ga4gh.htsget.v0.2rc+json; charset=utf-8
 
-## Authentication
+## Security
 
-Requests to the retrieval API endpoint may be authenticated by means of an OAuth2 bearer token included in the request headers, as detailed in [RFC 6750]. Briefly, the client supplies the header `Authorization: Bearer xxxx` with each HTTPS request, where `xxxx` is a private token. The mechanisms by which clients originally obtain their authentication tokens, and by which servers verify them, are currently beyond the scope of this specification. Servers may honor non-authenticated requests at their discretion.
+This data retrieval API enables the retrieval of highly sensitive genomic data by means of a client/server model.  Effective security measures are essential to protect the integrity and confidentiality of these data.  
+
+All communication channels used to transmit sensitive information, including security-critical information and genomic data, MUST BE protected using Transport Level Security (TLS) version 1.2 or later, as specified in [RFC 5246]. 
```

Comment:
Minor editorial: `…MUST be…`

Reply:
-1 on the change to require TLS.  When running the service only on localhost we do not enforce TLS.  We use this mode when deploying the htsget server into customer VMs to access their own data (eg, VM running samtools <-> localhost htsget server <-> TLS <-> GCS)

Reply:
From web browser's point of view (thinking web-based genome browser) it's better to have everything on https rather than some content on http and some on https.

Reply:
@kemp-ggh the text below says the security mechanisms only need to be implemented 'in accordance with the policies enforced by the applicable data holder', so if you can claim that the data holder does not require security for localhost, then that is your clause for not using TLS I think.

Reply:
@tk2: The “security mechanisms” that the text below is talking about are checking that “[identities] have been authenticated and [their] access authorised”, i.e., OAuth or the like. I don't see how that could be read to be related to the preceding paragraph's hard requirement for TLS.

On the mailing list @kemp-ggh suggested “All **public** communications channels”, and below the author @bakerdb suggests “All **open** communications channels”. Seems to me that something like one of those changes is the way to go for readably allowing “non-TLS in private” should we wish to (cf Mike's note on the mailing list).